tests/framework/extensions/rbac/verify.go

```diff
@@ -0,0 +1,261 @@
+func VerifyGlobalRoleBindingsForUser(t *testing.T, user *management.User, adminClient *rancher.Client) {
+	query := url.Values{"filter": {"userName=" + user.ID}}
+	grbs, err := adminClient.Steve.SteveType("management.cattle.io.globalrolebinding").List(query)
+	require.NoError(t, err)
+	assert.Equal(t, 1, len(grbs.Data))
+}
+
+// VerifyUserCanListCluster validates a user with the required global permissions are able to/not able to list the clusters in rancher server
+func VerifyUserCanListCluster(t *testing.T, client, standardClient *rancher.Client, clusterID, role string) {
+	clusterList, err := standardClient.Steve.SteveType(clusters.ProvisioningSteveResourceType).ListAll(nil)
+	require.NoError(t, err)
+
+	clusterStatus := &apiV1.ClusterStatus{}
+	err = v1.ConvertToK8sType(clusterList.Data[0].Status, clusterStatus)
+	require.NoError(t, err)
+
+	if role == restrictedAdmin {
+		adminClusterList, err := client.Steve.SteveType(clusters.ProvisioningSteveResourceType).ListAll(nil)
+		require.NoError(t, err)
+		assert.Equal(t, (len(adminClusterList.Data) - 1), len(clusterList.Data))
+	}
+	assert.Equal(t, 1, len(clusterList.Data))
+	actualClusterID := clusterStatus.ClusterName
+	assert.Equal(t, clusterID, actualClusterID)
+
+}
+
+// VerifyUserCanListProject validates a user with the required cluster permissions are able/not able to list projects in the downstream cluster
+func VerifyUserCanListProject(t *testing.T, client, standardClient *rancher.Client, clusterID, role, adminProjectName string) {
+	projectListNonAdmin, err := projects.ListProjectNames(standardClient, clusterID)
+	require.NoError(t, err)
+	projectListAdmin, err := projects.ListProjectNames(client, clusterID)
+	require.NoError(t, err)
+
+	switch role {
+	case roleOwner, restrictedAdmin:
+		assert.Equal(t, len(projectListAdmin), len(projectListNonAdmin))
+		assert.Equal(t, projectListAdmin, projectListNonAdmin)
+	case roleMember:
+		assert.Equal(t, 0, len(projectListNonAdmin))
+	case roleProjectOwner, roleProjectMember:
+		assert.Equal(t, 1, len(projectListNonAdmin))
+		assert.Equal(t, adminProjectName, projectListNonAdmin[0])
+	}
+}
+
+// VerifyUserCanCreateProjects validates a user with the required cluster permissions are able/not able to create projects in the downstream cluster
+func VerifyUserCanCreateProjects(t *testing.T, client, standardClient *rancher.Client, clusterID, role string) {
+	memberProject, err := standardClient.Management.Project.Create(projects.NewProjectConfig(clusterID))
+	switch role {
+	case roleOwner, roleMember, restrictedAdmin:
+		require.NoError(t, err)
+		log.Info("Created project as a ", role, " is ", memberProject.Name)
+		require.NoError(t, err)
+		actualStatus := fmt.Sprintf("%v", memberProject.State)
+		assert.Equal(t, activeStatus, strings.ToLower(actualStatus))
+	case roleProjectOwner, roleProjectMember:
+		require.Error(t, err)
+		errStatus := strings.Split(err.Error(), ".")[1]
+		rgx := regexp.MustCompile(`\[(.*?)\]`)
```

Comment:
Since this is used in multiple functions, can we not define this as a global variable?